app/models/cdr/table.rb

```diff
@@ -53,15 +53,59 @@ def archive
 
   def self.add_partition
     transaction do
-      self.execute_sp("SELECT * FROM sys.cdr_createtable(2)")  ##try to create table for previous month
-      self.execute_sp("SELECT * FROM sys.cdr_createtable(1)")  ##try to create table for current month
-      self.execute_sp("SELECT * FROM sys.cdr_createtable(0)") ##for next month
+      today = Date.today
+      create_table_for(today.prev_month)
+      create_table_for(today)
+      create_table_for(today.next_month)
+    end
+  end
+
+  def self.create_table_for(date_from)
```

Comment:
@dmitry-sinina please check this out.

Do we need to rewrite this functions, as well?
- sys.cdr_reindex
- sys.cdrtable_tgr_reload